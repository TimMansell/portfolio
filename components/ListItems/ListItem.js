import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';

import Icon from 'components/Icon';
import Labels from '../Labels';

import { useIsFlexGapSupported } from 'hooks/useFlexGap';

import styles from './ListItem.module.scss';

export const ListItem = ({ item }) => {
  const hasFlexGap = useIsFlexGapSupported();

  const labels = [
    {
      type: 'tertiary',
      items: item.list,
    },
  ];

  const itemClasses = classnames(styles.item, {
    [styles.flexGap]: hasFlexGap,
  });

  return (
    <div className={itemClasses}>
      <div className={styles.icon}>
        <Icon name={[item.icon.family, item.icon.name]} size="sm" />
      </div>
      <div className={styles.content}>
        <h3 className={styles.title}>{item.title}</h3>
        <Labels labels={labels} type="tertiary" size="lg" centered />
      </div>
    </div>
  );
};

ListItem.propTypes = {
  item: PropTypes.shape({
    icon: PropTypes.object.isRequired,
    title: PropTypes.string.isRequired,
    list: PropTypes.array.isRequired,
  }).isRequired,
};

export default ListItem;
