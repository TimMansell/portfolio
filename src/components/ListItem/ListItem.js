import React from 'react';
import PropTypes from 'prop-types';

import Icon from '../Icon';

export const ListItem = ({item}) => {
	return <div className="divider">
		<Icon name={[item.icon.family, item.icon.name]} size="md" padded />
		<ul className="skills__list">
			{item.list.map((item, i) =>
				<li className="skills__item" key={i}>{item}</li>
			)}
		</ul>
	</div>;
}

ListItem.propTypes = {
	item: PropTypes.object.isRequired
};

export default ListItem;