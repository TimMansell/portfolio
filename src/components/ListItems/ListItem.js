import React from 'react';
import PropTypes from 'prop-types';

import Icon from 'components/Icon';
import Labels from '../Labels';

import styles from './ListItem.module.scss';

export const ListItem = ({ item }) => {
  return (
    <div className={styles.item}>
      <div className={styles.icon}>
        <Icon name={[item.icon.family, item.icon.name]} size="sm" />
      </div>
      <div className={styles.content}>
        <h3 className={styles.title}>{item.title}</h3>
        <Labels items={item.list} type="tertiary" size="lg" centered />
      </div>
    </div>
  );
};

ListItem.propTypes = {
  item: PropTypes.shape({
    icon: PropTypes.object.isRequired,
    title: PropTypes.string.isRequired,
    list: PropTypes.array.isRequired,
  }).isRequired,
};

export default ListItem;
