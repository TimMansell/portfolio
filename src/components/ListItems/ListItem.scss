@import 'variables';

/* stylelint-disable-next-line selector-class-pattern */
.list-item {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 15px;

  @media (min-width: $bp-sm) {
    margin-bottom: 0;
  }

  &::after {
    content: '';
    margin-top: 20px;
    width: 60%;
    border-bottom: 1px solid rgba(0, 0, 0, 0.2);
    box-shadow: 0 1px rgba(255, 255, 255, 0.05);

    @media (min-width: $bp-sm) {
      content: none;
    }
  }

  &:last-of-type {
    &::after {
      content: none;
    }
  }

  &__list {
    list-style-type: none;
    font-size: 20px;
    text-shadow: -1px -1px rgba(0, 0, 0, 0.2);

    @media (min-width: $bp-sm) {
      font-size: 18px;
    }
  }

  &__item {
    margin-bottom: 5px;
  }
}
