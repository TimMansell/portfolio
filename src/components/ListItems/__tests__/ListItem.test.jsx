import { it, describe, expect } from 'vitest';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import ListItem from '../ListItem';

const defaultProps = {
  item: {
    title: 'title',
    list: ['list name 1', 'list name 2'],
    icon: {
      name: 'css3-alt',
      family: 'fab',
    },
  },
};

describe('ListItem', () => {
  it('should render my component', () => {
    // eslint-disable-next-line
    const wrapper = shallow(<ListItem {...defaultProps} />);
  });

  it('should match snapshot', () => {
    const snapshot = renderer.create(<ListItem {...defaultProps} />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });
});
