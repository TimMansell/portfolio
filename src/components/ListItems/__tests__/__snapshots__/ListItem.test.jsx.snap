// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ListItem > should match snapshot 1`] = `
<div
  className="_item_21adc7"
>
  <div
    className="_icon_21adc7"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-css3-alt fa-w-12 _icon_585b29 _iconSm_585b29"
      data-icon="css3-alt"
      data-prefix="fab"
      focusable="false"
      role="img"
      style={{}}
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M0 32l34.9 395.8L192 480l157.1-52.2L384 32H0zm313.1 80l-4.8 47.3L193 208.6l-.3.1h111.5l-12.8 146.6-98.2 28.7-98.8-29.2-6.4-73.9h48.9l3.2 38.3 52.6 13.3 54.7-15.4 3.7-61.6-166.3-.5v-.1l-.2.1-3.6-46.3L193.1 162l6.5-2.7H76.7L70.9 112h242.2z"
        fill="currentColor"
        style={{}}
      />
    </svg>
  </div>
  <div
    className="_content_21adc7"
  >
    <h3
      className="_title_21adc7"
    >
      title
    </h3>
    <ul
      className="_labels_591528 _isLarge_591528 _isCentered_591528"
      data-test="labels"
    >
      <li
        className="_label_103219 _isTertiary_103219 _isLarge_103219"
      >
        list name 1
      </li>
      <li
        className="_label_103219 _isTertiary_103219 _isLarge_103219"
      >
        list name 2
      </li>
    </ul>
  </div>
</div>
`;
